var/spack/repos/builtin/packages/unifyfs/package.py

```diff
@@ -16,55 +16,50 @@ class Unifyfs(AutotoolsPackage):
-    url      = "https://github.com/LLNL/UnifyFS/releases/download/v0.9.0/unifyfs-0.9.0.tar.gz"
+    url      = "https://github.com/LLNL/UnifyFS/releases/download/v0.9.2/unifyfs-0.9.2.tar.gz"
     maintainers = ['CamStan']
 
     version('develop', branch='dev')
+    version('0.9.2', sha256='7046625dc0677535f5d960187cb2e2d58a6f8cfb4dc6a3604f825257eb0891aa')
     version('0.9.1', sha256='2498a859cfa4961356fdf5c4c17e3afc3de7e034ad013b8c7145a622ef6199a0')
-    version('0.9.0', sha256='e6c73e22ef1c23f3141646aa17058b69c1c4e526886771f8fe982da924265b0f')
+    version('0.9.0', sha256='e6c73e22ef1c23f3141646aa17058b69c1c4e526886771f8fe982da924265b0f', deprecated=True)
 
     variant('auto-mount', default='True', description='Enable automatic mount/unmount in MPI_Init/Finalize')
     variant('hdf5', default='False', description='Build with parallel HDF5 (install with `^hdf5~mpi` for serial)')
     variant('fortran', default='False', description='Build with gfortran support')
-    variant('mdhim', default='False', description='Enable MDHIM build options')
     variant('pmi', default='False', description='Enable PMI2 build options')
     variant('pmix', default='False', description='Enable PMIx build options')
     variant('spath', default='True', description='Use spath library to normalize relative paths')
 
-    depends_on('autoconf',  type='build')
-    depends_on('automake',  type='build')
-    depends_on('libtool',   type='build')
-    depends_on('m4',        type='build')
-    depends_on('pkgconfig', type='build')
+    depends_on('autoconf',        type='build')
+    depends_on('automake',        type='build')
+    depends_on('automake@1.15:',  type='build', when='@0.9.2:')
+    depends_on('libtool',         type='build')
+    depends_on('m4',              type='build')
+    depends_on('pkgconfig',       type='build')
 
     # Required dependencies
-    depends_on('gotcha@1.0.3:', when='@0.9.1:')
-    depends_on('mochi-margo@0.4.3')
-    depends_on('mercury@1.0.1+bmi+sm')
+    depends_on('gotcha@1.0.3:')
+    depends_on('mercury@1.0.1+bmi', when='@:0.9.1')
+    depends_on('mochi-margo@0.4.3', when='@:0.9.1')
+    depends_on('mochi-margo',       when='@0.9.2:')
     depends_on('mpi')
     depends_on('openssl')
 
     # Optional dependencies
     depends_on('hdf5', when='+hdf5')
-    depends_on('leveldb', when='@0.9.1:+mdhim')
-    depends_on('spath', when='@0.9.1:+spath')
+    depends_on('libfabric fabrics=rxm,sockets,tcp', when="^mercury@2:+ofi")
+    depends_on('libfabric@1.9.1', when="^mochi-margo@0.4.3 ^mercury+ofi")
+    depends_on('mercury@1.0.1', when='^mochi-margo@0.4.3')
+    depends_on('spath~mpi', when='+spath')
 
-    # Required dependencies for older versions
-    depends_on('flatcc', when='@:0.9.0')
-    depends_on('gotcha@0.0.2', when='@:0.9.0')
-    depends_on('leveldb', when='@:0.9.0')
-
-    patch('unifyfs-sysio.c.patch', when='@0.9.1')
-
-    conflicts('^mercury~bmi')
+    conflicts('^mercury~bmi~ofi')
```

Comment:
(question) Note that this package would accept `^mercury~bmi+ofi` or `^mercury+bmi~ofi` - was that your intent? If both are needed, you'll need separate `conflicts` declarations (in that case though you can also add these as constraints to the relevant `depends_on` declaration and this will be avoided).

Reply:
Yes, my intent was that we'll accept `^mercury+bmi~ofi`, `^mercury~bmi+ofi`, or `^mercury+bmi+ofi`.

With version `0.9.1`, we defaulted to using bmi, since our ofi support wasn't stable. However, we now default to ofi, but we'll fallback to bmi if ofi is not available. So the only time we won't work is with `^mercury~ofi~bmi`.

I do have `depends_on('mercury@1.0.1+bmi', when='@:0.9.1')` in this package, so perhaps I should add back something like `conflicts('^mercury~bmi', when='@0.9.1')`?

Reply:
> However, I do have depends_on('mercury@1.0.1+bmi', when='@:0.9.1') in this package, so perhaps I should add back something like conflicts('^mercury~bmi', when='@0.9.1')?

That wouldn't be necessary: `depends_on('mercury@1.0.1+bmi', when='@:0.9.1')` will enforce `^mercury+bmi` for version `0.9.1`.